homeassistant/components/sensor/miflora.py

```diff
@@ -0,0 +1,135 @@
+"""
+Support for Xiaomi Mi Flora BLE plant sensor.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.miflora/
+"""
+from datetime import timedelta
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.helpers.entity import Entity
+import homeassistant.helpers.config_validation as cv
+from homeassistant.util import Throttle
+from homeassistant.const import CONF_MONITORED_CONDITIONS, CONF_NAME
+
+
+REQUIREMENTS = ['miflora==0.1.4']
+
+LOGGER = logging.getLogger(__name__)
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=900)
+CONF_MAC = "mac"
+CONF_FORCE_UPDATE = 'force_update'
+DEFAULT_NAME = "Mi Flora"
+
+# Sensor types are defined like: Name, units
+SENSOR_TYPES = {
+    'temperature': ['Temperature', '°C'],
+    'light': ['Light intensity', 'lux'],
+    'moisture': ['Moisture', '%'],
+    'conductivity': ['Conductivity', 'µS/cm'],
+}
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_MAC): cv.string,
+    vol.Required(CONF_MONITORED_CONDITIONS):
+        vol.All(cv.ensure_list, [vol.In(SENSOR_TYPES)]),
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_FORCE_UPDATE, default=False): cv.boolean,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the MiFlora sensor."""
+    from miflora import miflora_poller
+
+    poller = miflora_poller.MiFloraPoller(config.get(CONF_MAC))
+    force_update = config.get(CONF_FORCE_UPDATE)
+
+    devs = []
+
+    for parameter in config[CONF_MONITORED_CONDITIONS]:
+        name = SENSOR_TYPES[parameter][0]
+        unit = SENSOR_TYPES[parameter][1]
+
+        prefix = config.get(CONF_NAME)
+
+        if len(prefix) > 0:
+            name = prefix + " " + name
```

Comment:
Consistency with the rest of the app.

On Aug 30, 2016 10:39 PM, "Open Home Automation" <notifications@github.com>
wrote:

> In homeassistant/components/sensor/miflora.py
> <https://github.com/home-assistant/home-assistant/pull/3053#discussion_r76923497>
> :
>
> > +    """Setup the MiFlora sensor."""
> > +    from miflora import miflora_poller
> > +
> > +    poller = miflora_poller.MiFloraPoller(config.get(CONF_MAC))
> > +    force_update = config.get(CONF_FORCE_UPDATE)
> > +
> > +    devs = []
> > +
> > +    for parameter in config[CONF_MONITORED_CONDITIONS]:
> > +        name = SENSOR_TYPES[parameter][0]
> > +        unit = SENSOR_TYPES[parameter][1]
> > +
> > +        prefix = config.get(CONF_NAME)
> > +
> > +        if len(prefix) > 0:
> > +            name = prefix + " " + name
>
> We can, but is this really better? What's the problem of a simple
> concatenation?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/home-assistant/home-assistant/pull/3053/files/8577a75a0be3c7217532fb72802fb9aeb262dc2f#r76923497>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AC2fZZB0y_c2MYB8O_ceqrxRXIeCK1b5ks5qlQVkgaJpZM4JwJvn>
> .
>